src/components/AttachmentCarousel/createInitialState.js

```diff
@@ -50,7 +51,7 @@ function createInitialState(props) {
 
     const page = _.findIndex(attachments, (a) => a.source === props.source);
     if (page === -1) {
-        throw new Error('Attachment not found');
+        Navigation.dismissModal();
     }
```

Comment:
Do we need to `return` after this so that `props.onNavigate` doesn't still get called?

Reply:
Ya looks like this is causing this regression: https://github.com/Expensify/App/issues/23964

Reply:
@dukenv0307 interesting - I did this and it seemed to work fine for me: https://github.com/Expensify/App/pull/23988

So I returned an object, not `undefined`

Reply:
Thanks for offering creating a PR, but I already created one 😅  ^ feel free to review if you'd like

Reply:
Hey @Beamanator and @dukenv0307,

I've noticed we've opted to remove the error throw in this section. While I understand this might be an attempt to address the issue of deleting an image, we need to recall that it was previously flagged as a bug when the carousel closed upon an image deletion ([Issue #18451](https://github.com/Expensify/App/issues/18451)). Reintroducing this behavior might not be the best move.

The original intent of the `createInitialState`—as the name suggests—was to execute just once upon the carousel's opening. Its purpose was to display the "Ooops" screen if a user tries to open an unlocatable attachment. In my opinion, presenting this error screen offers a more insightful user experience compared to abruptly closing the modal, which may come off as if the user's action was disregarded.

Can we reconsider this change and perhaps retain the error throw?